src/main/java/com/cloudogu/smeagol/wiki/infrastructure/GitClient.java

```diff
@@ -0,0 +1,150 @@
+package com.cloudogu.smeagol.wiki.infrastructure;
+
+import com.cloudogu.smeagol.Account;
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.eclipse.jgit.revwalk.RevCommit;
+import org.eclipse.jgit.transport.CredentialsProvider;
+import org.eclipse.jgit.transport.RefSpec;
+import org.eclipse.jgit.transport.UsernamePasswordCredentialsProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.net.URL;
+import java.util.Iterator;
+import java.util.Optional;
+
+import static java.util.Collections.singleton;
+
+public class GitClient implements AutoCloseable {
+
+    private static final Logger LOG = LoggerFactory.getLogger(GitClient.class);
+
+    private final Account account;
+    private final File repository;
+    private final URL remoteUrl;
+    private final String branch;
+
+    private Git git;
+
+    public GitClient(Account account, File repository, URL remoteUrl, String branch) {
+        this.account = account;
+        this.repository = repository;
+        this.remoteUrl = remoteUrl;
+        this.branch = branch;
+    }
+
+    public void refresh() throws GitAPIException, IOException {
+        if ( repository.exists() ) {
+            pullChanges();
+        } else {
+            createClone();
+        }
+    }
+
+    private Git open() throws IOException {
+        if (git == null) {
+            git = Git.open(repository);
+        }
+        return git;
+    }
+
+    public File file(String path) {
+        return new File(repository, path);
+    }
+
+    public Optional<RevCommit> lastCommit(String path) throws IOException, GitAPIException {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this method to throw at most one checked exception instead of: java.io.IOException, org.eclipse.jgit.api.errors.GitAPIException [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://ecosystem.cloudogu.com/sonar/coding_rules#rule_key=squid%3AS1160)